clients/src/main/java/org/apache/kafka/common/requests/ConsumerGroupHeartbeatRequest.java

```diff
@@ -59,6 +60,11 @@ public Builder(ConsumerGroupHeartbeatRequestData data, boolean enableUnstableLas
 
         @Override
         public ConsumerGroupHeartbeatRequest build(short version) {
+            if (version == 0 && data.subscribedTopicRegex() != null) {
+                throw new UnsupportedVersionException("The cluster does not support regular expressions resolution " +
+                    "on version " + version + ". It must be upgraded to version >= 1 to allow to subscribe to a " +
+                    "SubscriptionPattern.");
```

Comment:
nit: I wonder whether we should precise that the version is the version of the ConsumerGroupHeartbeat API. Users may confused it with the version of the client or the server. What do you think?

Reply:
makes sense , done